### ISSUE TITLE
Gists: Bring back the (editable) title (and descriptions)

### ISSUE THREAD

**Issue by mhulse** [isaacs/github#194]
### TL;DR:

Allow for gist titles and descriptions, like the old gist system, that are editable and independent from the gist itself.

---
### Long version:

I don't know about others, but I REALLY miss the old gist naming system.

In the old version, you could give your gist a title. You could also edit this title (independently from the gist) at any time.

These days, the "title" is the name of the first file in a gist. For an example, look at anyone's gist landing page. Here's mine:

https://gist.github.com/mhulse

Maybe it's just me, but I really don't like seeing the file name as the main title. It's not very helpful when scanning with eyes.

Additionally, what happens when your gist has several different files? Not all gists are one file gists. The title is created from the first file in the the gist, yet that file may not be the important one out of the group (IIRC, gist files are automatically sorted by file name A-Z).

Some workarounds:
1. Name the file something like `This is my example of doing X for Y language.xml`.
2. When there are multiple files to a gist, do the same as above, and add numbers to the start of each file name: `01. This is my example of doing X for Y language.xml`, `02. Here's and example of PHP doing X with language.xml.php`.

While I love the improvements from the old gist system in general, I really miss being able to give my gists titles.

**Side note:** Descriptions are now a part of the gist. This means that you have to edit the whole gist to modify description. ~~Doing so creates a new revision. I'd personally prefer the description to be separated from the gist as to avoid revisions if I'm just tweaking description info.~~ **UPDATE:** I was wrong. While the description is a part of the gist, editing **just** the description **does NOT create a new revision**.
### GitHub support reply:

> Thanks for your feedback! We're always working to improve GitHub and we consider every suggestion we receive. I've added your idea about the editable Gist titles to the Feature Request List™ for the team to see.

IIRC, I sent them the TL;DR version of my request. :+1:

**Comment by simonc** [comment]
When using a multi-file gist having it named after the first file is really annoying. A gist name `Gemfile` doesn't really make sense :)

**Comment by Winsto** [comment]
plus one for this. I also really like @patcon's readme.md idea.

As I'm here adding this in Aug 2015 then I'd like to echo @asantos comment; any update?

**Comment by mechanoid** [comment]
recently added license files to my gists, which i feel is a good idea, but the current title naming compromises the list overview totally :/ So please consider this issue a look, thx

**Comment by danijar** [comment]
+1

**Comment by fincher** [comment]
Seems this is still an issue.  My current remedy is to basically have a "README" type file that just follows the format "a_set_of_files_for_accompishing_X" which then becomes the title since "a_" is almost always alphabetically before any of the other files included.  At the very least, the option to choose which file would be the title instead of just the first would be a step in the right direction.

**Comment by lsloan** [comment]
Similar to @fincher's suggestion, in gists with multiple files I've created filenames with leading special characters (e.g., space, underscore, etc.).  That would make them appear first in the list and appear as the name of the gist.  It's not ideal, but it works well enough for the moment.

**Comment by system-addict** [comment]
I found this thread by way of looking for 'the ultimate character' to use. I'm using an underscore as I do in my windows folder directories for **_assets**.  I imagine it would also generate less trouble if the need ever comes to parse the gists.

Glad to know others are thinking the same and yes, proposed solutions should maybe be bumped as README.md would seem to fit nicely in community standards.

**Comment by JohnL4** [comment]
Now my gist is named "App.config". Great. >:(

**Comment by cben** [comment]
Gists currently have an editable description / subtitle, right?
And it appears (in smaller font) in https://gist.github.com/USERNAME list.
Is this simply about the fact the bolder file name as "title" is distracting?

**Comment by mhulse** [comment]
@cben you raise a good point. Maybe the title of this issue could be better. I am open to suggestions. :)

Some things that bother me:

1. The gist is named after the first file in the gist.
1. When editing a title of a gist, it creates a new version in the revision history (though, it says "no changes").
1. Skimming gists, on the user index page, is hard and not a good experience.

If you had used the old system I think you would understand better where I am coming from.

**Comment by Winsto** [comment]
@cben the description is editable, but there isn't an editable title. What appears under the `https://gist.github.com/USERNAME` list is `USERNAME/firstFileInGistForThatGist`, which is the major pain as @mhulse says.

To get a meaningful list of gist titles you end up having a file along the lines of `01_What_Would_Like_As_My_Gist_Tutle.md` or something like that. For example I have [01_Useful_PowerShell_Things.md](https://gist.github.com/Winsto/5245618db6158e9f096d). What we would really like is to have a gist title `Useful PowerShell Things` a brief description 'bits of PowerShell I've found useful' and then just start adding `.ps1` files as examples underneath.

**Comment by alexgittens** [comment]
Come on, github. It's been three years since this comment thread opened. What's the hold up in moving to a more reasonable naming system for gists?

**Comment by StephenOrJames** [comment]
Ultimately I would also like to see an editable title. However, a workaround that I've found to this issue is creating an empty file at the top of the Gist with your desired title as that file's name. This works, because the first file name is used as the title. Unfortunately, you'd have to put up with the extra space that would take up, but it's arguably worth it.

**Comment by aleclarson** [comment]
I wouldn't mind hearing the Github team's rationale regarding this decision.

Is this by design? What are the benefits of not having custom titles?

The default value can still be the name of the first file, rather than requiring a title exist.

Please throw us a bone. 😄 🐶

**Comment by surdu** [comment]
Before, you could name a file that started with a space in order to have a way to name the gist. 

Now even that is gone ...

**Comment by kopp** [comment]
@StephenOrJames : It seems like the files are sorted automatically by name and the name of the first one is used as title of the gist.  So when I added a file `Readme.md` my gist was called `Readme.md` -- so I renamed it to `z_Readme.md` :wink:

**Comment by solongtony** [comment]
This is a really obnoxious issue, which makes multiple file gists almost unusable.  I just tried the workaround of adding a new file named with a title. The file showed up, at the top of the file list, but the name of the gist didn't change :(
 
I would expect this kind of thing from Atlassian's Stash, but not from GitHub.

**Comment by simonc** [comment]
It's been more than 4 years and no answers from the Github team 😢 . Do those issues serve any purpose? 😕

I have no idea of the complexity behind the gists system but from the outside it doesn't seem that complicated to add back a simple name to it. I'd be really valuable to understand why it was removed and the rational for not adding it back.

**Comment by b4dnewz** [comment]
alexa, this is so sad

**Comment by xahon** [comment]
+1

**Comment by SiddharthShyniben** [comment]
For now, I guess one way to do it would be to create a file for your title and start it with a space character, and put invisible characters in that file (here's an invisible character: `   ‎‎‎‎‎‎‎‎‎‎‎‎‎‎‎‎‎‎   `)

**Comment by sheldonhull** [comment]
Maybe folks watching this could please add some response/mention to github to raise awareness here as well in social media. Mentioned this thread. https://twitter.com/sheldon_hull/status/1407368544010924033?s=20

**Comment by Levi-Lesches** [comment]
...or join [the official GitHub feedback repo](https://github.com/github/feedback/discussions/categories/general-feedback) and start a thread there. Reminder that GitHub staff doesn't monitor these pages.
